File: common/QuadraticEquation/quadratic.py
import math

def solve_quadratic(a, b, c):
    discriminant = b ** 2 - 4 * a * c

    if discriminant > 0:
        root1 = (-b + math.sqrt(discriminant)) / (2 * a)
        root2 = (-b - math.sqrt(discriminant)) / (2 * a)
        return (root1, root2)
    elif discriminant == 0:
        root = -b / (2 * a)
        return (root,)
    else:
        return "两个虚根"

a = float(input("请输入 a, b, c 的值: "))
b = float(input())
c = float(input())

result = solve_quadratic(a, b, c)

if isinstance(result, tuple):
    if len(result) == 2:
        print(f"两个实数根: {result[0]}, {result[1]}")
    else:
        print(f"一个实数根: {result[0]}")
else:
    print(result)

File: common/QuadraticEquation/quadratic.c
#include <stdio.h>
#include <math.h>

int main() {
    double a, b, c;
    printf("请输入 a, b, c 的值: ");
    scanf("%lf %lf %lf", &a, &b, &c);

    double discriminant = b * b - 4 * a * c;

    if (discriminant > 0) {
        double root1 = (-b + sqrt(discriminant)) / (2 * a);
        double root2 = (-b - sqrt(discriminant)) / (2 * a);
        printf("两个实数根: %lf, %lf\n", root1, root2);
    } else if (discriminant == 0) {
        double root = -b / (2 * a);
        printf("一个实数根: %lf\n", root);
    } else {
        printf("两个虚根\n");
    }

    return 0;
}